src/main/kotlin/org/rust/lang/core/types/RustTypeParameterType.kt

```diff
@@ -16,6 +16,8 @@ class RustTypeParameterType(val parameter: RustTypeParamElement) : RustTypeBase(
     override fun getNonStaticMethodsIn(project: Project): Sequence<RustFnElement> =
         getTraitsImplementedIn(project).flatMap { it.traitMethodMemberList.asSequence() }
 
+    override fun substitute(map: Map<RustTypeParameterType, RustType>): RustType = map[this] ?: this
+
```

Comment:
And would be the direct benefits of `RustTypeParameterTag`? I fail to see any at the moment. Perhaps we'd better add it later, when/if we understand why exactly do we need it?